src/vs/workbench/electron-browser/workbench.ts

```diff
@@ -975,11 +975,9 @@ export class Workbench implements IPartService {
 		if (event.reason === ShutdownReason.RELOAD) {
 			const workspace = event.payload;
 
-			// We are transitioning into a workspace from an empty workspace or folder workspace
-			// As such we want to migrate UI state from the current workspace to the new one. Since
-			// many components write to storage only on shutdown, we register a shutdown listener
-			// very late to be called as the last one.
-			if (isWorkspaceIdentifier(workspace) && !this.contextService.hasMultiFolderWorkspace()) {
+			// We are transitioning into a workspace from an empty workspace or workspace, and
+			// as such we want to migrate UI state from the current workspace to the new one. 
+			if (isWorkspaceIdentifier(workspace)) {
```

Comment:
@sandy081 fyi the migration will now also run when going from one workspace to another, we need this to support a smooth transition when saving a workspace.